### PR TITLE
[vcpkg] Fix a TODO in the dependency resolution and some warnings

### PR DESCRIPTION
**Pull request by cbezault**
Clean up the way default features are handled as well as some warnings.

### CODE REVIEW

**Comment by cbezault** [comment]
@ras0219-msft could you add the tests for #7471 to this, I think it has been fixed (maybe not even by this PR).

File: toolsrc/src/vcpkg/dependencies.cpp [hunk 1]
```diff
@@ -300,6 +236,72 @@ namespace vcpkg::Dependencies
...
+            info.build_edges.emplace(feature, dep_list);
+        }
+        else
+        {
+            bool requires_qualified_resolution = false;
+            for (const Dependency& dep : *qualified_deps)
+            {
+                if (dep.qualifier.empty())
+                {
+                    FullPackageSpec fspec({dep.depend.name, m_spec.triplet()}, dep.depend.features);
```
Comment (cbezault) [inline toolsrc/src/vcpkg/dependencies.cpp]: I need to take into account whether or not the dependency is installed here. If it is not I need to add the default features even if core is specified explicitly in the dependency.

File: toolsrc/src/vcpkg/dependencies.cpp [hunk 2]
```diff
@@ -300,6 +236,72 @@ namespace vcpkg::Dependencies
...
+        if (maybe_vars)
+        {
+            // Qualified dependency resolution is available
+            auto fullspec_list = filter_dependencies(
+                *qualified_deps, m_spec.triplet(), maybe_vars.value_or_exit(VCPKG_LINE_INFO));
+
+            for (auto&& fspec : fullspec_list)
+            {
+                const Cluster& dep = m_parent.get(fspec.package_spec);
+                auto&& defaults = dep.m_scfl.source_control_file->core_paragraph->default_features;
```
Comment (cbezault) [inline toolsrc/src/vcpkg/dependencies.cpp]: I need to take into account whether or not the dependency is installed here. If it is not I need to add the default features even if core is specified explicitly in the dependency.

**Comment by cbezault** [comment]
I'm going to add a test to catch the bug I would have introduced with this PR.

File: toolsrc/src/vcpkg/dependencies.cpp (hunk 2 above)
Comment (cbezault) [reply toolsrc/src/vcpkg/dependencies.cpp:266]: We actually handle this in `Cluster::create_install_info` so this is a non-issue.

**Comment by JackBoosY** [comment]
@cbezault I think you should merge the master branch to this branch.

**Pull request closed without merging**